FirebaseAuthExchange/Sources/Models/ExchangeServiceModels.swift

```diff
@@ -60,3 +60,66 @@ struct ExchangeOIDCTokenRequest: Encodable {
     self.authCodeCredentials = authCodeCredentials
   }
 }
+
+// MARK: - Response Models
+
+/// Response that encapsulates an Auth Exchange access token.
+///
+/// This is the return value for an `ExchangeInstallationAuthTokenRequest` or
+/// `ExchangeCustomTokenRequest`.
+struct ExchangeTokenResponse: Decodable, Equatable {
+  /// A container for a Firebase  access token and the duration of time until it expires.
+  struct AuthExchangeToken: Decodable, Equatable {
+    /// A signed [JWT](https://tools.ietf.org/html/rfc7519) containing claims that identify a user.
+    let accessToken: String
+
+    /// The duration of time until the `accessToken` expires, approximately relative to the time
+    /// this response was received.
+    let timeToLive: ProtobufDuration
+
+    enum CodingKeys: String, CodingKey {
+      case accessToken
+      case timeToLive = "ttl"
+    }
+  }
+
+  /// An Auth Exchange access token that can be used to access Firebase services.
+  let token: AuthExchangeToken
+}
+
+/// Model of a `google.protobuf.Duration`, which represents a time span.
+///
+/// A Protocol Buffer
+/// [`Duration`](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#duration)
+/// represents a signed, fixed-length span of time represented as a count of seconds and fractions
+/// of seconds at nanosecond resolution. It is independent of any calendar and concepts like "day"
+/// or "month".
+struct ProtobufDuration: Equatable {
```

Comment:
Haha we were just discussing whether to implement a type for `Duration`, or just do a quick cast on the spot. Do you intend to expose this to the public (or something similar), or is this just for the network (service) layer?

Reply:
This is just for the network layer so keeping it as a `String` like you did in [`AuthExchangeTokenP`](https://github.com/firebase/firebase-android-sdk/blob/1817518ab0d4286e95028126ea946c125c6153a6/firebase-authexchange/src/main/kotlin/com/google/firebase/authexchange/network/Routes.kt#L36) is also something we should consider for the Swift SDK. I liked the idea of having access to the seconds and nanos components for testing purposes (vs. floating-point values) but I do need to convert it to a [`TimeInterval`](https://developer.apple.com/documentation/foundation/timeinterval) (i.e., a `Double`) eventually in order to expose it as `expirationDate` of type [`Date`](https://developer.apple.com/documentation/foundation/date) in the public [`AuthExchangeToken`](https://github.com/firebase/firebase-ios-sdk/blob/8cdc6fef63310e9c861689dbb0f0d0829ef22c86/FirebaseAuthExchange/Sources/AuthExchangeToken.swift#L16-L25): https://github.com/firebase/firebase-ios-sdk/blob/8cdc6fef63310e9c861689dbb0f0d0829ef22c86/FirebaseAuthExchange/Sources/AuthExchangeToken.swift#L16-L25

Planning to use [`init(timeIntervalSince1970: TimeInterval)`](https://developer.apple.com/documentation/foundation/date/1780353-init) for that.

Reply:
Yeah in the[ App Check Android implementation](https://github.com/firebase/firebase-android-sdk/blob/df79e47b6d2fc14552f8ac9053bb6a190d325642/appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/AppCheckTokenResponse.java#L38), we just kept it as a String, but I'm fine with either approach.